bob/src/cluster/quorum.rs

```diff
@@ -272,18 +272,64 @@ impl Cluster for Quorum {
     }
 
     async fn exist(&self, keys: &[BobKey]) -> Result<Vec<bool>, Error> {
-        let keys_by_nodes = group_keys_by_nodes(&self.mapper, keys);
-        debug!(
-            "EXIST Nodes for fan out: {:?}",
-            &keys_by_nodes.keys().flatten().collect::<Vec<_>>()
-        );
         let len = keys.len();
         let mut exist = vec![false; len];
-        for (nodes, (keys, indexes)) in keys_by_nodes {
-            let cluster_results = LinkManager::exist_on_nodes(&nodes, &keys).await;
-            for result in cluster_results.into_iter().flatten() {
-                for (&r, &ind) in result.inner().iter().zip(&indexes) {
-                    exist[ind] |= r;
+
+        // filter local keys
+        let (indices, local_keys) = filter_local_keys(keys, &self.mapper);
+        // end
+        debug!("local keys {:?}", local_keys);
+        debug!("local indices {:?}", indices);
```

Comment:
2 logs should be merged into one. Also they should be `trace!`, because they print the raw request data